src/lightning/fabric/strategies/fsdp.py

```diff
@@ -255,30 +251,32 @@ def setup_module_and_optimizers(
         module = self.setup_module(module)
         return module, optimizers
 
-    def setup_module(self, module: Module) -> "FullyShardedDataParallel":
+    def setup_module(self, module: Module) -> Module:
         """Wraps the model into a
         :class:`~torch.distributed.fsdp.fully_sharded_data_parallel.FullyShardedDataParallel` module."""
-        from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel
-
-        if "auto_wrap_policy" in self._fsdp_kwargs and any(
-            isinstance(mod, FullyShardedDataParallel) for mod in module.modules()
-        ):
-            # If model is already wrapped, we need to avoid sending the `auto_wrap_policy`
-            del self._fsdp_kwargs["auto_wrap_policy"]
-        wrapped_module = FullyShardedDataParallel(
-            module=module,
-            cpu_offload=self.cpu_offload,
-            mixed_precision=self.mixed_precision_config,
-            sharding_strategy=self.sharding_strategy,
-            device_id=self.root_device.index,
-            **self._fsdp_kwargs,
-        )
+        from torch.distributed.fsdp import FullyShardedDataParallel
+
+        if any(isinstance(mod, FullyShardedDataParallel) for mod in module.modules()):
```

Comment:
Does `module.modules()` also include the root module or only its children? If it includes both, feel free to resolve, otherwise we need to check the root module as well :)

Reply:
Yes, modules() returns all submodules recursively, including the root module itself.